File: dotnet-backend/projekatFlutter/Models/MaintenanceRecord.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class MaintenanceRecord:IEntity<int>
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int VehicleId { get; set; }

        [Required]
        [MaxLength(50)]
        public string MaintenanceType { get; set; } = string.Empty;

        [Required]
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        [Column(TypeName = "decimal(10,2)")]
        public decimal? Cost { get; set; }

        [MaxLength(3)]
        public string Currency { get; set; } = "BAM";

        public int? OdometerReading { get; set; }

        [MaxLength(100)]
        public string? ServiceProvider { get; set; }

        [MaxLength(100)]
        public string? PerformedBy { get; set; }

        public DateTime? ScheduledDate { get; set; }
        public DateTime? CompletedDate { get; set; }

        [Required]
        [MaxLength(30)]
        public string Status { get; set; } = "Scheduled";

        public DateTime? NextServiceDate { get; set; }
        public int? NextServiceOdometer { get; set; }

        [MaxLength(1000)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        [ForeignKey("VehicleId")]
        public Vehicle Vehicle { get; set; } = null!;
    }
}

File: frontend/lib/models/maintenance_record.dart
class MaintenanceRecord {
  final int id;
  final int vehicleId;
  final String maintenanceType;
  final String description;
  final double? cost;
  final String currency;
  final int? odometerReading;
  final String? serviceProvider;
  final String? performedBy;
  final DateTime? scheduledDate;
  final DateTime? completedDate;
  final String status;
  final DateTime? nextServiceDate;
  final int? nextServiceOdometer;
  final String? notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  MaintenanceRecord({
    required this.id,
    required this.vehicleId,
    required this.maintenanceType,
    required this.description,
    this.cost,
    required this.currency,
    this.odometerReading,
    this.serviceProvider,
    this.performedBy,
    this.scheduledDate,
    this.completedDate,
    required this.status,
    this.nextServiceDate,
    this.nextServiceOdometer,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });
  factory MaintenanceRecord.fromJson(Map<String,dynamic> json)
  {
    return MaintenanceRecord(id: json['Id'], vehicleId: json['VehicleId'], maintenanceType: json['MaintenanceType'], description: json['Description'],cost:json['Cost'] ,currency: json['Currency'],odometerReading: json['OdometerReading'],serviceProvider: json['ServiceProvider'],performedBy: json['PerformedBy'] ,scheduledDate: json['ScheduledDate'],completedDate: json['CompletedDate'],status: json['Status'],nextServiceDate: json['NextServiceDate'],nextServiceOdometer: json['NextServiceOdometer'],notes: json['Notes'] ,createdAt: json['CreatedAt'], updatedAt: json['UpdatedAt']);
  }
  Map<String, dynamic> toJson() {
  return {
    'id': id,
    'vehicleId': vehicleId,
    'maintenanceType': maintenanceType,
    'description': description,
    'cost': cost,
    'currency': currency,
    'odometerReading': odometerReading,
    'serviceProvider': serviceProvider,
    'performedBy': performedBy,
    'scheduledDate': scheduledDate?.toIso8601String(),
    'completedDate': completedDate?.toIso8601String(),
    'status': status,
    'nextServiceDate': nextServiceDate?.toIso8601String(),
    'nextServiceOdometer': nextServiceOdometer,
    'notes': notes,
    'createdAt': createdAt.toIso8601String(),
    'updatedAt': updatedAt.toIso8601String(),
  };
}
}